public/js/reader.js

```diff
@@ -1,408 +1,601 @@
+    }
+
+    // Update page numbers in the paginator
+    const newVal = Reader.showingSinglePage
+        ? Reader.currentPage + 1
+        : `${Reader.currentPage + 1} + ${Reader.currentPage + 2}`;
+    $(".current-page").each((_i, el) => $(el).html(newVal));
+
+    Reader.updateImageMap();
+    Reader.currentPageLoaded = true;
+    $("#i3").removeClass("loading");
+};
+
+Reader.updateImageMap = function () {
+    // update imagemap with the w/h parameters we obtained
+    const img = $("#img")[0];
+    const mapWidth = img.width / 2;
+    const mapHeight = img.height;
+
+    $("#leftmap").attr("coords", `0,0,${mapWidth},${mapHeight}`);
+    $("#rightmap").attr("coords", `${mapWidth + 1},0,${img.width},${mapHeight}`);
+};
+
+Reader.goToPage = function (page, fromHistory = false) {
+    Reader.previousPage = Reader.currentPage;
+    Reader.currentPage = Math.min(Reader.maxPage, Math.max(0, +page));
+
+    Reader.showingSinglePage = false;
+    if (Reader.doublePageMode && Reader.currentPage > 0 && Reader.currentPage < Reader.maxPage) {
+        // composite an image and use that as the source
+        img1 = Reader.loadImage(Reader.pages[Reader.currentPage], Reader.canvasCallback);
+        img2 = Reader.loadImage(Reader.pages[Reader.currentPage + 1], Reader.canvasCallback);
+    } else {
+        // In single view, just use the source URLs as is
+        $("#img").attr("src", Reader.pages[Reader.currentPage]);
+        Reader.showingSinglePage = true;
+    }
+
+    // Preload N images, in the future this can be turned into a setting
+    let preloadNext = 2;
+    let preloadPrev = 1;
+
+    if (Reader.doublePageMode) { preloadNext *= 2; preloadPrev *= 2; }
+    if (Reader.mangaMode) { [preloadNext, preloadPrev] = [preloadPrev, preloadNext]; }
+
+    for (let i = 1; i <= preloadNext; i++) {
+        if (Reader.currentPage + i > Reader.maxPage) { break; }
+        Reader.loadImage(Reader.pages[Reader.currentPage + i], null);
+    }
+    for (let i = 1; i <= preloadPrev; i++) {
+        if (Reader.currentPage - i < 0) { break; }
+        Reader.loadImage(Reader.pages[Reader.currentPage - i], null);
+    }
+
+    Reader.applyContainerWidth();
+
+    Reader.currentPageLoaded = false;
+    // display overlay if it takes too long to load a page
+    setTimeout(() => {
+        if (!Reader.currentPageLoaded) { $("#i3").addClass("loading"); }
+    }, 500);
+
+    // update full image link
+    $("#imgLink").attr("href", Reader.pages[Reader.currentPage]);
+
+    // Send an API request to update progress on the server
+    genericAPICall(`api/archives/${Reader.id}/progress/${Reader.currentPage + 1}`, "PUT", null, "Error updating reading progress!", null);
+
+    // scroll to top
+    window.scrollTo(0, 0);
+
+    // Update url to contain all search parameters, and push it to the history
+    if (!fromHistory) { window.history.pushState(null, null, `?id=${Reader.id}&p=${page + 1}`); }
```

Comment:
This sets `&p=0` somehow when opening a reader URL without the `p` query parameter.  

Looks like it's also possible to +/-1 the parameter when on the first/last pages:  
![dfrg](https://user-images.githubusercontent.com/8237712/116123421-44522600-a6c3-11eb-872f-99a41c7c9a2a.gif)  
This doesn't really hamper functionality and I probably wouldn't have noticed it without the issue with progress reported above 😅

Reply:
It was a typo, during rebases a variable name wasn't ported over. Fixed now.